liblangutil/SourceLocation.h

```diff
@@ -62,6 +62,13 @@ struct SourceLocation
 		return start <= _other.start && _other.end <= end;
 	}
 
+	inline bool contains(int _pos) const
+	{
+		if (!hasText())
```

Comment:
Maybe we don't need to check for `source`?

Reply:
done (not pushed yet)